### ISSUE TITLE
Issue 08 — `Binomial` natural-space `getgradlogpartition` → NaN for large logits

### ISSUE THREAD

**Issue by docxology** [ReactiveBayes/ExponentialFamily.jl#297]
# Issue 08 — `Binomial` natural-space `getgradlogpartition` → NaN for large logits

**Component:** `src/distributions/binomial.jl:101-104`
**Severity:** MAJOR
**Type:** Numerical / silent wrong inference

## Summary
The natural-space gradient of the log-partition (`= E[x]`, the posterior-mean
quantity) uses `n·exp(η)/(1+exp(η))`, which becomes `NaN` for `η ≳ 709`:

```julia
getgradlogpartition(::NaturalParametersSpace, ::Type{Binomial}, ntrials) = (η) -> begin
    (η₁,) = unpack_parameters(Binomial, η)
    return SA[ntrials*exp(η₁)/(one(η₁)+exp(η₁))]
end
```
`exp(1000.0)/(1+exp(1000.0))` → `NaN`.

## Evidence / repro (Julia 1.12.6)
```julia
getgradlogpartition(NaturalParametersSpace(), Binomial, 100)([1000.0])
# => [NaN]      (n·logistic(1000) would be 100.0)
```

## Impact
Silent `NaN` for a `Binomial` factor whose logit is driven large (confident/degenerate
posteriors). Inconsistent with the package's own stable `NaturalToMean` (`logistic(η₁)`),
so the two definitions of the same quantity diverge exactly in the confident regime.

## Proposed fix
```julia
return SA[ntrials * logistic(η₁)]
```
(`StatsFuns.logistic` is already imported at binomial.jl:6.)

**Comment by bvdmitri** [comment]
Fixed in #302 and released in v2.6.0 (see the `[2.6.0]` CHANGELOG entry): `Binomial` natural-space `getgradlogpartition` now uses `logistic` instead of an `exp`-ratio, so large logits no longer produce `NaN` (`src/distributions/binomial.jl`). Regression test added in 0a442f6.
